### ISSUE TITLE
Duplication of resistance genes in typing report

### ISSUE THREAD

**Issue by samuell** [Clinical-Genomics/microSALT#185]
**Describe the bug**

After the introduction of microSALT 4.0.0, a customer reports that resistance genes are listed two times after each other in each sample, as exemplified in the screenshot below, where four resistance genes are reported twice resulting in 8 rows:

**Expected behavior**
Each resistance gene should be reported only once per sample.

**Screenshots**

![image](https://github.com/user-attachments/assets/b4056fb1-4cea-4090-9b3b-362f99881b75)

**Software version (please complete the following information):**
- microSALT: 4.0.0

**Additional context**
- (On a more positive note the customer was happy that OXA-48 is reported correctly here and not as OXA-566 (See #149 ) )

**Comment by samuell** [comment]
It seems now that what is different between the stage and production environment now is that we for some reason have set of files named `all.*` in the resistances folder, which naturally contains duplicates of the ones in the other folders. This one is not listed in the `config` file, and is not there in the stage environment, so I think it is created by mistake, possibly by some file we manually created while debugging.

**Comment by samuell** [comment]
I tried copying in the `all.*` files from the production resistances folder to the stage one, and now I could reproduce the duplication.

**Comment by samuell** [comment]
The question then remain as to why these duplicates have not shown up before. This seems to be due to a small change in the elimination of duplicates that we implemented as part of the 4.0.0 release, and which is restored in 4fdc651

**Comment by samuell** [comment]
In summary:

1. The basic reason why the duplicates appear is that there are all.* files with duplicates, in the resistances directory in prod.
2. The reason these appear only now is because we changed a line in microSALT's duplicate handling, which seemed to be doing the wrong thing, but apparently managed to remove these duplicates.
  - I have tried to revert this change in 4fdc651 and then the duplicates go away again. 

So apparently there are two ways to fix the duplicate problem:

1. Just removing all the .* files from the resistances directory (they don't seem to belong there anyway?)
2. To restore the changed line according to the mentioned commit (4fdc651) in a patch.
